src/read.c

```diff
@@ -2013,6 +2030,108 @@ static avifBool avifParseToneMappedImageBox(avifGainMapMetadata * metadata, cons
 }
 #endif // AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP
 
+#if defined(AVIF_ENABLE_EXPERIMENTAL_SAMPLE_TRANSFORM)
+// bit_depth is assumed to be 2 (32-bit).
+static avifResult avifParseSampleTransformTokens(avifROStream * s, avifSampleTransformExpression * expression)
+{
+    uint8_t tokenCount;
+    AVIF_CHECK(avifROStreamRead(s, &tokenCount, /*size=*/1)); // unsigned int(8) token_count;
+    AVIF_CHECKERR(avifArrayCreate(expression, sizeof(expression->tokens[0]), tokenCount), AVIF_RESULT_OUT_OF_MEMORY);
```

Comment:
Reject `tokenCount == 0` now? We should not pass 0 as the `initialCapacity` argument to `avifArrayCreate()`.

Alternatively, pass `AVIF_MAX(1, tokenCount)` as the `initialCapacity` argument to `avifArrayCreate()` and let the `avifSampleTransformExpressionIsValid()` call reject it.